modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/helper/DLPortletInstanceSettingsHelper.java

```diff
@@ -107,6 +118,79 @@ else if (!portletName.equals(DLPortletKeys.DOCUMENT_LIBRARY) &&
 		return entryColumns;
 	}
 
+	public Folder getRootFolder() throws PortalException {
+		DLPortletInstanceSettings dlPortletInstanceSettings =
+			_dlRequestHelper.getDLPortletInstanceSettings();
+
+		String rootFolderExternalReferenceCode =
+			dlPortletInstanceSettings.getRootFolderExternalReferenceCode();
+
+		if (Validator.isBlank(rootFolderExternalReferenceCode)) {
+			return null;
+		}
+
+		ThemeDisplay themeDisplay = _dlRequestHelper.getThemeDisplay();
+
+		Group selectedGroup =
+			GroupLocalServiceUtil.getGroupByExternalReferenceCode(
+				dlPortletInstanceSettings.
+					getSelectedGroupExternalReferenceCode(),
+				themeDisplay.getCompanyId());
+
+		return new LiferayFolder(
+			DLFolderLocalServiceUtil.getDLFolderByExternalReferenceCode(
```

Comment:
Isn't it better to use com.liferay.document.library.kernel.service.DLAppLocalServiceUtil#getFolderByExternalReferenceCode in order to not use the low level services?

Reply:
Yes, you are right this the right method to use. But after checking again I bumped into this method, now you know why I'm not using it. Or do you think it is better to implement it?

```java
	public default Folder getFolderByExternalReferenceCode(
			String externalReferenceCode)
		throws PortalException {

		throw new UnsupportedOperationException(
			"This method needs to be implemented");
	}
```

Reply:
This is the default implementation of the Interface (to avoid break possible custom DocumentRepository  implementations)

All our specific implementations have implement it.

Reply:
I've found the problem. The class `LiferayLocalRepository` didn't implement the method. I've added 2 commits:

* [LPD-33264 Fix - add two missing methods implementation](https://github.com/liferay-content-management/liferay-portal/pull/6412/commits/9931c943800924526b265ecba64b35de082a4642)
* [LPD-33264 Use the appropriate service](https://github.com/liferay-content-management/liferay-portal/pull/6412/commits/34949002b602689a8f34faeabdf3f7a02f55e289)